src/pyscipopt/scip.pxi

```diff
@@ -975,6 +975,130 @@ cdef class Constraint:
         return (self.__class__ == other.__class__
                 and self.scip_cons == (<Constraint>other).scip_cons)
 
+cdef class Decomposition:
+    """Base class holding a pointer to SCIP decomposition"""
+    def __init__(self):
+        self.create()
+    @staticmethod
+    cdef create(SCIP_DECOMP* scip_decomp):
+        if scip_decomp == NULL:
+            raise Warning("cannot create Constraint with SCIP_CONS* == NULL")
+        decomp = Decomposition()
+        decomp.scip_decomp = scip_decomp
+        return decomp
+
+    def isOriginal(self):
```

Comment:
We're trying to make sure new functionality isn't added without having tests. So for all available functions you're adding please put them into a simple test (one test can cover multiple functions)

Reply:
Will add more tests & docs when I get a moment, sorry. This one got put on the backburner due to other stuff at work.

For our use case, we have a modeling layer on top of PySCIPOpt that basically uses dataframe-like syntax to create groups of constraints using a table that characterizes our variables. So the goal is to break `.groupby` statements into subproblems where each of the groups can be solved ~in parallel. We have some combinatorial constraints as well, so decompositions seem like a natural place to reduce the number of combinations to check.